src/Core/Settings.cpp

```diff
@@ -1918,6 +1918,13 @@ See also:
 For single JOIN in case of identifier ambiguity prefer left table
 )", IMPORTANT) \
     \
+DECLARE(BoolAuto, query_plan_join_swap_table, {}, R"(
```

Comment:
cc @devcrafter

Reply:
@vdimir Honestly, the setting looks weird. IMHO, the setting should just enable or disable the automatic mode to choose the table. But to use this setting for swapping tables in JOIN clause - looks weird - user should just rewrite the query itself.

Reply:
It's also useful for testing, as I want to ensure that any pipeline we build is valid and can be executed. 

Another reason is that when the automatically inferred side is incorrect, and the user need an ad-hoc solution to specify the correct one (however, as you mentioned, rewriting is also an option and should work for most use cases).

But in ideal case the setting is used only in tests and rarely by user set to `false` to have legacy behavior.